### PR TITLE
Compressor effect fixes

### LINKED ISSUE #13237: Compressor effect: Adjust Makeup Time constant
### Bug Description

The Compressor introduced in https://github.com/mixxxdj/mixxx/pull/12523
Use the input RMS value to generate a makeup gain. This is required to allow fading and outros. 
Unfortunately this value does not scale with the compressor timing nor the engine buffer.   

### Version

2.5~alpha

### OS

_No response_

### PR DESCRIPTION
**Pull request by fonsargo**
Fixes #13237

### CODE REVIEW

File: src/effects/backends/builtin/compressoreffect.cpp [hunk 1]
```diff
@@ -152,75 +194,62 @@ void CompressorEffect::processChannel(
...
-        // smoothing
-        makeUpGainState = kMakeUpAttackCoeff * makeUp + (1 - kMakeUpAttackCoeff) * makeUpGainState;
+    // smoothing
+    makeUpGainState = pState->makeUpCoeff * rmsGain + (1 - pState->makeUpCoeff) * makeUpGainState;
 
-        pState->previousMakeUpGain = makeUpGainState;
-        SampleUtil::applyGain(pOutput, makeUpGainState, numSamples);
-    }
+    pState->makeUpGainState = makeUpGainState;
+    SampleUtil::applyGain(pGainBuffer, 1 / makeUpGainState, numSamples);
```
Comment (daschuer) [inline src/effects/backends/builtin/compressoreffect.cpp:233]: I think we need to ramp here as well:
```suggestion
    SampleUtil::applyRampingGain(pGainBuffer, 1 / pState->makeUpGainState, 1/makeUpGainState, numSamples);
    pState->makeUpGainState = makeUpGainState;
```

Comment (fonsargo) [reply src/effects/backends/builtin/compressoreffect.cpp]: Fixed!

**Review by daschuer — APPROVED** [review]
LGTM and sounds good. Do your wish to rebase the PR to main and squash the fixup commits.

**Comment by fonsargo** [comment]
Rebased and squashed

**Comment by acolombier** [comment]
@daschuer merge?

**Comment by daschuer** [comment]
yes.

**Pull request merged**